Source/Core/AudioCommon/Mixer.cpp

```diff
@@ -88,6 +87,14 @@ unsigned int Mixer::MixerFifo::Mix(short* samples, unsigned int numSamples,
                         (static_cast<u64>(m_input_sample_rate_divisor) * 1000);
     low_watermark = std::min(low_watermark, MAX_SAMPLES / 2);
 
+    // Throw away extra audio when the writer exits fast forward.
+    if (should_resync && numLeft > low_watermark)
+    {
+      numLeft = m_numLeftI = low_watermark;
+      indexR = (indexW - 2 * low_watermark) & INDEX_MASK;
+      ASSERT(numLeft == static_cast<float>(((indexW - indexR) & INDEX_MASK) / 2));
```

Comment:
What is this verifying? A small comment would help in case this assert actually triggered.

Reply:
Normally the code computes `numLeft = ((indexW - indexR) & INDEX_MASK) / 2`. When ending fast-forward and throwing away all but `low_watermark` frames of audio, we assign `numLeft = low_watermark` then rewrite `indexR` to `low_watermark` frames (or 2 * that samples) before the end of available audio. This assertion is just to check that I got my math right, and the `indexR` I write has the correct number of samples left; I copied the formula from the initial point we calculate `float numLeft`. (I'd hope that `2 * low_watermark` is always less than `INDEX_MASK`, if not the current code would already break?)

Should I keep this assert or not? I'm not sure how to comment it.

Reply:
I'd keep it, and add something like "not enough samples to continue"